substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/c/libc/LibCBase.java

```diff
@@ -24,18 +24,53 @@
  */
 package com.oracle.svm.core.c.libc;
 
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Method;
 import java.nio.file.Path;
+import java.util.Arrays;
 import java.util.List;
 
+import org.graalvm.nativeimage.ImageSingletons;
+
 public interface LibCBase {
 
     String PATH_DEFAULT = "<default>";
 
+    static <T extends AnnotatedElement> boolean isProvidedInCurrentLibc(T element) {
+        LibCBase currentLibC = ImageSingletons.lookup(LibCBase.class);
+        Libc targetLibC = element.getAnnotation(Libc.class);
+        return targetLibC == null || Arrays.asList(targetLibC.value()).contains(currentLibC.getClass());
```

Comment:
Maybe add a comment comment that this policy causes the use of `@Libc` allows to tag parts of libraries that are exclusive to a specific Libc implementation.

Reply:
This is a good point, added a javadoc explaining how the check if a type is provided works.